src/Illuminate/Foundation/helpers.php

```diff
@@ -124,6 +124,28 @@ function app_path($path = '')
     }
 }
 
+if(! function_exists("array_flatten_keeping_keys")) {
+    /**
+     * @param $array
+     * @param string $prefix
+     * @return array
+     * Thanks
+     * https://stackoverflow.com/questions/9546181/flatten-multidimensional-array-concatenating-keys/9546215#9546215
+     */
+    function array_flatten_keeping_keys($array, $prefix = '') {
+        $result = array();
+        foreach($array as $key=>$value) {
```

Comment:
Why duplicated code? just use the method you defined

Reply:
yup! I forgot to go back to this. My bad. I made the helper first then realized after going to write a test for it that I need to really make the Arr::keeping_keys 
by the time I was done with that lost track of going back here.